### ISSUE TITLE
Add express rate limit to server

### ISSUE THREAD

**Issue by tylermaran** [rysolv/rysolv#109]
This one should be pretty straight forward. 

Add a rate-limit package (https://www.npmjs.com/package/express-rate-limit) to the [server file](https://github.com/rysolv/rysolv/blob/master/server/index.js).

RIght now I think it should be a general rate limit (no need for route by route limits).
I think that 25 requests per minute would be a reasonable limit.

**Comment by tylermaran** [comment]
Eventually I would like to have something where GET requests have a high limit, but POST requests
But it's a little more compliacted since all requests are technically going through graphql (all POST req).

**Comment by wilhsie** [comment]
Saw your msg in webdev disc, i've been looking to hop into an open source proj to gain open source exp.  Can I take this one?

**Comment by tylermaran** [comment]
@wilhsie yea of couse! 

For this one you could probably just make the changes without having to get things running locally. 
But if you want to get some more experience, then we can work on getting your local project set up as well.

The easiest way for me to walk you through environment setup (and send some env variables) is if you can hop on the discord server ([invite link](https://discord.gg/axCNYAXk))

Generally the process will be: 
- Fork the repo
- Check out a branch from `dev` formatted with your initials / task. Something like: 
    - `git checkout dev`, `git pull`
    - `git checkout -b TM/fix-deleted-user-link`
- Follow instructions in the readme to get project running
    - if you run into problems here, please let me know so we can improve the readme
- Make the changes, test a couple scenarios
- Open a pull reqest with your branch against rysolv/dev
- I'll take a look, test it myself, and either propose changes or merge it in
- Done!

**Comment by wilhsie** [comment]
thanks for this, I joined the disc earlier today as @mangekbro
I'm currently going thru the readme and I ran into an issue when seeding the database:
`
[ERROR] AssertionError [ERR_ASSERTION]: ifError got unwanted exception: connect ECONNREFUSED 127.0.0.1:5432
    at module.exports (/Users/wyvren/Documents/code/opensrc/rysolv/node_modules/db-migrate/lib/commands/helper/assert.js:9:14)
    at /Users/wyvren/Documents/code/opensrc/rysolv/node_modules/db-migrate/lib/commands/down.js:21:14
    at /Users/wyvren/Documents/code/opensrc/rysolv/node_modules/db-migrate/connect.js:17:7
    at /Users/wyvren/Documents/code/opensrc/rysolv/node_modules/db-migrate/lib/driver/index.js:95:9
    at Client._connectionCallback (/Users/wyvren/Documents/code/opensrc/rysolv/node_modules/db-migrate-pg/index.js:710:7)
    at Client._handleErrorWhileConnecting (/Users/wyvren/Documents/code/opensrc/rysolv/node_modules/pg/lib/client.js:296:19)
    at Client._handleErrorEvent (/Users/wyvren/Documents/code/opensrc/rysolv/node_modules/pg/lib/client.js:306:19)
    at Connection.emit (node:events:394:28)
    at Socket.reportStreamError (/Users/wyvren/Documents/code/opensrc/rysolv/node_modules/pg/lib/connection.js:53:12)
    at Socket.emit (node:events:394:28)
    at emitErrorNT (node:internal/streams/destroy:193:8)
    at emitErrorCloseNT (node:internal/streams/destroy:158:3)
    at processTicksAndRejections (node:internal/process/task_queues:83:21)
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1132:16)
`

have you seen this happen before?

**Comment by tylermaran** [comment]
@wilhsie Hmm this is probably because it can't connec to your local postgres instance. 

Was this when you tried to run the seed script?

And did you take a look at the postgres setup here: https://github.com/rysolv/rysolv/blob/master/server/README.md

**Comment by wilhsie** [comment]
Oh duh, sorry about that.  Yes this happened when I ran npm run seed, but ofc it errored out because I didn't have postgres configured.  I'm up and running now

**Comment by tylermaran** [comment]
Hey @wilhsie, how's this one coming? Running into any more issues I could help with?

**Comment by wilhsie** [comment]
@tylermaran hi, I added the express rate limiter to the server file, but I'm a bit unfamiliar with how to test it.  Was gonna continue reading up on graphql today and try coming up with something..  But I'm also open to any suggestions

**Comment by tylermaran** [comment]
Go ahead and put in a PR so I can look at the changes. 

It should be pretty straightforward. The rate limit is just a middleware that sits in front of any of the other routes and only allows a certain amount of traffic through. 

You shouldn't need to do anything with GraphQl in this story. Eventually I'd like to add some more fine tuned rate limiting based on the route. But for now, I think just a general 25 / minute limit. 

If you want to test it out, you could try hitting the endpoint manually through `localhost:3000/graphql`
On local development it should bring up the graphql playground. And you can test out calls. 

Something like this will do. And just spam click it and see what kind of error comes back.
```
query {
      oneIssue(id: "${id}") {
        __typename
        ... on Issue {
          name
        }
        ... on Error {
          message
        }
      }
```

**Comment by tylermaran** [comment]
closed by #127
